src/bench/check_connectblock.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2016-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+
+#include <addresstype.h>
+#include <interfaces/chain.h>
+#include <kernel/cs_main.h>
+#include <script/interpreter.h>
+#include <sync.h>
+#include <test/util/setup_common.h>
+#include <validation.h>
+
+#include <cassert>
+#include <utility>
+#include <vector>
+
+std::pair<CBlock, std::unique_ptr<CBlockIndex>> CreateTestBlock(TestChain100Setup* test_setup)
+{
+    Chainstate& chainstate{test_setup->m_node.chainman->ActiveChainstate()};
+
+    // Create a block with (500*4) schnorr signatures to verify
+    // Typical blocks have high number of txs with few inputs
+    const int num_txs = 500;
+    const int num_inputs = 4;
```

Comment:
Could we use an existing block from Mainnet () - to be sure we're measuring a representative case and not one skewed towards our preferences?
One with many schnorr sigs - but since we don't yet have one that only contains them, I don't think it makes sense to measure that (like we do here).

Reply:
They're always skewed, but if we use actual blocks as basis, at least we can't trick ourselves so easily.

Reply:
> They're always skewed, 

Not if we use randomization.

> but if we use actual blocks as basis, at least we can't trick ourselves so easily.

We are just tricking ourselves differently.

After a brief discussion with @Eunovo and @josibake elsewhere I think there should be 3 separate benchmarks: No Schnorr Blocks, Random-mix Blocks and only schnorr blocks. This allows to see the impact of changes on each of these scenarios and the random mix blocks should cover what we currently see on chain.

Reply:
> They're always skewed - Not if we use randomization.

We always introduce bias, it's unavoidable. Let's try to come up with something that minimizes our own preference of how we wish the blocks looked like (as in this PR where the block only contained Schnorrs). We can avoid that by checking the actual usages and relying on them.

> I think there should be 3 separate benchmarks: No Schnorr Blocks, Random-mix Blocks and only schnorr blocks. 

This will likely minimize the biases that we introduce - especially, as mentioned before, if we rely on existing blocks, instead of an artificial setup that are skewed towards the settings we desire to measure.

Reply:
I've updated the PR to use 3 blocks
- all schnorr
- 50% schnorr, 50% ecdsa
- all ecdsa

If we used one block from Mainnet here, we would optimize for that block. This setup of 3 blocks is more balanced. We can always do an actual IBD benchmark or use tools like [Benchcoin](https://github.com/bitcoin-dev-tools/benchcoin) cc @josibake 

Connecting random blocks from Mainnet is tricky. At best, I would only be able to use the Mainnet block as a template to create the blocks for the benchmark. I'm not sure it's worth it to do this.

Reply:
> we would optimize for that block

So what are we optimizing for now?